GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -15,8 +15,8 @@ internal interface IRevisionGraphRowProvider
     // The RevisionGraph contains all the basic structures needed to render the graph.
     public class RevisionGraph : IRevisionGraphRowProvider
     {
-        public const int MaxLanes = 40;
-        private const int _straightenLanesLookAhead = 1;
+        internal const int MaxLanes = 40;
+        private const int _straightenLanesLookAhead = 20;
```

Comment:
These two constants could be turned into settings, which might solve #8498, too.

Reply:
While I personally like options, I believe GE has too many options for most users so I will by default vote no to new options.
If this can be described well it could be added.

Reply:
> Instead of another panel, the column could be made resizable. I am happy with the dynamic resizing.

This could be the configuration for MaxLanes

Reply:
Can we detect automatically how many lanes required?

Reply:
> Can we detect automatically how many lanes required?

For this, it would not suffice to load the revisions: The graph segments must be created, which is an expensive operation. That's why it is done in chunks of at most 1500 rows on user request.

`MaxLanes` is a hard-coded limit. (I have not introduced it. I moved it here from `RevisionGraphColumnProvider` in the predecessor PR.)